conda_forge_tick/auto_tick.py

```diff
@@ -0,0 +1,736 @@
+    order :
+        Build order for this migrator
+    """
+    out: Dict[str, Set[str]] = {
+        "done": set(),
+        "in-pr": set(),
+        "awaiting-pr": set(),
+        "awaiting-parents": set(),
+        "bot-error": set(),
+    }
+
+    gx2 = copy.deepcopy(getattr(migrator, "graph", gx))
+
+    top_level = {node for node in gx2 if not list(gx2.predecessors(node))}
+    build_sequence = list(cyclic_topological_sort(gx2, top_level))
+
+    feedstock_metadata = dict()
+
+    import graphviz
+    from streamz.graph import _clean_text
+
+    gv = graphviz.Digraph()
+    for node, node_attrs in gx2.nodes.items():
+        attrs = node_attrs["payload"]
+        # remove archived from status
+        if attrs.get("archived", False):
+            continue
+        node_metadata: Dict = {}
+        feedstock_metadata[node] = node_metadata
+        nuid = migrator.migrator_uid(attrs)
+        for pr_json in attrs.get("PRed", []):
+            if pr_json and pr_json["data"] == frozen_to_json_friendly(nuid)["data"]:
+                break
+        else:
+            pr_json = None
+
+        # No PR was ever issued but the migration was performed.
+        # This is only the case when the migration was done manually
+        # before the bot could issue any PR.
+        manually_done = pr_json is None and frozen_to_json_friendly(nuid)["data"] in (
+            z["data"] for z in attrs.get("PRed", [])
+        )
+
+        buildable = not migrator.filter(attrs)
+        fntc = "black"
+        if manually_done:
+            out["done"].add(node)
+            fc = "#440154"
+            fntc = "white"
+        elif pr_json is None:
+            if buildable:
+                out["awaiting-pr"].add(node)
+                fc = "#35b779"
+            else:
+                out["awaiting-parents"].add(node)
+                fc = "#fde725"
+        elif "PR" not in pr_json:
+            out["bot-error"].add(node)
+            fc = "#000000"
+            fntc = "white"
+        elif pr_json["PR"]["state"] == "closed":
+            out["done"].add(node)
+            fc = "#440154"
+            fntc = "white"
+        else:
+            out["in-pr"].add(node)
+            fc = "#31688e"
+            fntc = "white"
+        if node not in out["done"]:
+            gv.node(
+                node,
+                label=_clean_text(node),
+                fillcolor=fc,
+                style="filled",
+                fontcolor=fntc,
+            )
+
+        # additional metadata for reporting
+        node_metadata["num_descendants"] = len(nx.descendants(gx2, node))
+        node_metadata["immediate_children"] = [
+            k
+            for k in sorted(gx2.successors(node))
+            if not gx2[k].get("payload", {}).get("archived", False)
+        ]
+        if pr_json and "PR" in pr_json:
+            # I needed to fake some PRs they don't have html_urls though
+            node_metadata["pr_url"] = pr_json["PR"].get("html_url", "")
```

Comment:
maybe send this to the feedstock page?

Reply:
Maybe make an improvement issue for this one.

Reply:
#732